python/cudf/cudf/core/series.py

```diff
@@ -529,22 +539,43 @@ def append(self, other, ignore_index=False):
         -------
         A new Series equivalent to self concatenated with other
         """
-        this = self
-        other = Series(other)
+        if verify_integrity not in (None, False):
+            raise NotImplementedError(
+                "verify_integrity parameter is not supported yet."
+            )
+
+        if isinstance(to_append, (list, tuple)):
+            to_concat = [self]
+            to_concat.extend(to_append)
+        else:
+            to_concat = [self, Series(to_append)]
+
+        to_concat = [Series(obj) for obj in to_concat]
```

Comment:
We shouldn't convert the objects to a Series, we should validate if they're a Series or a DataFrame and throw accordingly otherwise.
```
In [28]: pd.Series([1,2,3]).append(np.array([4]))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-28-86444cbfe73a> in <module>
----> 1 pd.Series([1,2,3]).append(np.array([4]))

~/miniconda3/lib/python3.7/site-packages/pandas/core/series.py in append(self, to_append, ignore_index, verify_integrity)
   2580             to_concat = [self, to_append]
   2581         return concat(
-> 2582             to_concat, ignore_index=ignore_index, verify_integrity=verify_integrity
   2583         )
   2584

~/miniconda3/lib/python3.7/site-packages/pandas/core/reshape/concat.py in concat(objs, axis, join, ignore_index, keys, levels, names, verify_integrity, sort, copy)
    279         verify_integrity=verify_integrity,
    280         copy=copy,
--> 281         sort=sort,
    282     )
    283

~/miniconda3/lib/python3.7/site-packages/pandas/core/reshape/concat.py in __init__(self, objs, axis, join, keys, levels, names, ignore_index, verify_integrity, copy, sort)
    355                     "only Series and DataFrame objs are valid".format(typ=type(obj))
    356                 )
--> 357                 raise TypeError(msg)
    358
    359             # consolidate

TypeError: cannot concatenate object of type '<class 'numpy.ndarray'>'; only Series and DataFrame objs are valid
```

Reply:
@kkraus14 While I was looking at this, this bug surfaced: https://github.com/rapidsai/cudf/issues/5599

I'll fix it as part of a separate PR and get that to be merged before making changes in this PR.